client/rpc/src/chain/mod.rs

```diff
@@ -185,6 +185,44 @@ where
 	) -> RpcResult<bool> {
 		Ok(self.subscriptions().cancel(id))
 	}
+
+	/// Finalized block headers subscription.
+	///
+	/// Returns headers of all blocks that have been finalized, rather than just the latest.
+	fn subscribe_new_finalized_block_headers(
+		&'static self,
```

Comment:
Forgot to mention - it only compiles if I add this explicit `'static` lifetime, which seems reasonable to me (how could the subscription outlive itself?).

The annotation is required because the stream captures `self` inside the call to `then` on line 207.

Any thoughts @AlastairHolmes ?

Reply:
Actually this is going to require some non-trivial changes - jsonrpc doesn't like the static lifetime annotation.